### PR TITLE
feat(inter-protocol)!: rename InterestRate to StabilityFee

### LINKED ISSUE #7588: Replace 'Interest Rate' with 'Stability Fee' in inter-protocol
## What is the Problem Being Solved?
Similarly to https://github.com/Agoric/agoric-sdk/issues/7099 - Need to change 'Interest Rate'. Product have opted for 'Stability Fee'.

https://github.com/Agoric/dapp-inter/issues/108 handles the UI changes but the scope of this ticket is to update it everywhere else in the code, tests, contracts, documentation (non-user facing) etc to align.

## Description of the Design
Likely a breaking change.

Replace 'Interest Rate' with 'Stability Fee' in code (might be InterestRate so StabilityFee is acceptable)

```[tasklist]
### Blocked by 
- [ ] https://github.com/Agoric/agoric-sdk/issues/7337
- [ ] https://github.com/Agoric/agoric-sdk/issues/7407
```

## Security Considerations
N/A
## Scaling Considerations
Unknown
## Test Plan
TBD

### PR DESCRIPTION
**Pull request by samsiegart**
fixes #7588

### CODE REVIEW

File: packages/inter-protocol/docs/governance.md [hunk 1]
```diff
@@ -51,16 +51,16 @@ the Inter Protocol Whitepaper, v0.8.
 | Governance Key     | Type              | WP? |
 | ------------------ | :---------------- | --- |
 | DebtLimit          | Amount            | Yes |
-| InterestRate       | Ratio             | Yes |
+| StabilityFee       | Ratio             | Yes |
 | LiquidationPadding | Ratio             |     |
 | LiquidationMargin  | Ratio             | Yes |
 | LiquidationPenalty | Ratio             | Yes |
-| LoanFee            | Ratio             | Yes |
+| MintFee            | Ratio             | Yes |
```
Comment (samsiegart) [inline packages/inter-protocol/docs/governance.md:58]: This was already changed everywhere but here

File: packages/inter-protocol/test/vaultFactory/snapshots/test-vaultFactory.js.md [hunk 2]
```diff
@@ -79,7 +79,11 @@ Generated by [AVA](https://avajs.dev).
               value: 101967213114754098360n,
             },
           },
-          interestRate: {
+          latestInterestUpdate: {
```
Comment (dckc) [inline packages/inter-protocol/test/vaultFactory/snapshots/test-vaultFactory.js.md]: Should `latestInterestUpdate` be renamed for similar reasons?

File: packages/inter-protocol/test/vaultFactory/test-storage.js [hunk 3]
```diff
@@ -48,7 +48,7 @@ test.only('storage keys', async t => {
     managerA,
     'asset',
     'mockChainStorageRoot.vaultFactory.managers.manager0',
-    ['compoundedInterest', 'interestRate', 'latestInterestUpdate'],
+    ['compoundedInterest', 'latestInterestUpdate', 'stabilityFee'],
```
Comment (dckc) [inline packages/inter-protocol/test/vaultFactory/test-storage.js]: likewise, should `compoundedInterest` be renamed?

File: packages/inter-protocol/test/vaultFactory/test-vaultFactory.js [hunk 4]
```diff
@@ -398,15 +398,14 @@ test('vaultFactory display collateral', async t => {
     liquidationMargin: makeRatio(105n, run.brand),
     stabilityFee: makeRatio(530n, run.brand, BASIS_POINTS),
     marketPrice: makeRatio(5_555_555n, run.brand, 1_000_000n, aeth.brand),
-    interestRate: makeRatio(100n, run.brand, 10000n, run.brand),
```
Comment (dckc) [inline packages/inter-protocol/test/vaultFactory/test-vaultFactory.js]: should the `stabilityFee` value be `100n` instead of `530n`?

**Review by dckc — COMMENTED** [review]
should `compoundedInterest` and `latestInterestUpdate` likewise be renamed?

File: packages/inter-protocol/test/vaultFactory/snapshots/test-vaultFactory.js.md (hunk 2 above)
Comment (samsiegart) [reply packages/inter-protocol/test/vaultFactory/snapshots/test-vaultFactory.js.md]: Done as discussed

File: packages/inter-protocol/test/vaultFactory/test-storage.js (hunk 3 above)
Comment (samsiegart) [reply packages/inter-protocol/test/vaultFactory/test-storage.js]: Done as discussed

File: packages/inter-protocol/test/vaultFactory/test-vaultFactory.js (hunk 4 above)
Comment (samsiegart) [reply packages/inter-protocol/test/vaultFactory/test-vaultFactory.js]: Looks like there was a pre-existing `stabilityFee` here colliding with the newly named `stabilityFee` (`interestRate`). It should have been the `mintFee` instead. Fixed.

File: packages/inter-protocol/src/vaultFactory/vaultDirector.js [hunk 5]
```diff
@@ -471,7 +471,7 @@ export const prepareVaultDirector = (
                     liquidationMargin: vm
                       .getGovernedParams()
                       .getLiquidationMargin(),
-                    stabilityFee: vm.getGovernedParams().getMintFee(),
+                    mintFee: vm.getGovernedParams().getMintFee(),
```
Comment (dckc) [inline packages/inter-protocol/src/vaultFactory/vaultDirector.js]: eek! that would have been bad.

**Comment by samsiegart** [comment]
@dckc I also renamed the `debtSnapshot.interest` property to `debtSnapshot.stabilityFee`, PTAL then I was planning to resolve conflicts after approval before merging

**Review by dckc — APPROVED** [review]
looks good.

**Pull request merged**